app/renderer/src/actions/Inspector.js

```diff
@@ -849,22 +879,72 @@ export function setAwaitingMjpegStream(isAwaiting) {
   };
 }
 
+export function uploadGesturesFromFile(fileList) {
+  return async (dispatch) => {
+    const fileReaderPromise = fileList.map((file) => {
+      const reader = new FileReader();
+      return new Promise((resolve) => {
+        reader.onload = (event) =>
+          resolve({
+            name: file.name,
+            content: event.target.result,
+          });
+        reader.readAsText(file);
+      });
+    });
+    const gestures = await Promise.all(fileReaderPromise);
+    const invalidGestures = [];
+    const parsedGestures = [];
+    gestures.forEach((gestureFile) => {
+      try {
+        const gesture = JSON.parse(gestureFile.content);
+        const isValid = gesture.name && gesture.actions.map(isValidGesture).every(Boolean);
+        if (!isValid) {
+          invalidGestures.push(gestureFile.name);
+        } else {
+          gesture.description = gesture.description || '';
+          parsedGestures.push(_.omit(gesture, ['id']));
+        }
+      } catch (e) {
+        invalidGestures.push(gestureFile.name);
```

Comment:
maybe it would be easier for users to understand what the problem is if we also add the actual failure description. the current description (e.g. `invalid format`) is not very helpful.

Reply:
I have now added detailed error logs with the reason for invalid gesture format.

<img width="628" alt="Screenshot 2024-06-14 at 10 29 54 PM" src="https://github.com/appium/appium-inspector/assets/20136913/9e856d5e-e8b7-416b-924e-df7c9611c312">